modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,49 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+/* NOTE FOR handle_taur_sprites(): This can be converted to handle things like digi sprites, vox, etc. I just didn't do it because I don't want to refactor the entire
+system for this one pr. ~niko */
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Only runs on /item/clothing/under/ and /item/clothing/suit/, with an overriding proc for suits. Will not fire at all if the clothing has been defined to have all taur types.
+ * Then checks an associative list of icon_states for all 3 DMI files for it's type (uniform or suit) to see if the icon_state of the clothing matches any, and if it does,
+ * it will assign the relevant bitflag. Due to how bitflags work, checking for the bitflag isn't needed, we can just apply it regardless.
+ *
+ * Has seperate logic for suits, as they use a different set of DMIs as of now.
+ */
+/obj/item/clothing/proc/handle_taur_sprites()
+
+	if (!(istype(src, /obj/item/clothing/under))) //remove this if we ever make a taur sprite for anything else, this is for performance
+		return
+
+	if (mutant_variants & STYLE_TAUR)
+		return
+
+	if (GLOB.naga_taur_uniform_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_SNAKE
+	if (GLOB.horse_taur_uniform_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_HOOF
+	if (GLOB.pawed_taur_uniform_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_PAW
+
+/obj/item/clothing/suit/handle_taur_sprites()
+
+	if (mutant_variants & STYLE_TAUR)
+		return
+
+	if (GLOB.naga_taur_suit_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_SNAKE
+	if (GLOB.horse_taur_suit_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_HOOF
+	if (GLOB.pawed_taur_suit_sprites[icon_state])
+		mutant_variants |= STYLE_TAUR_PAW
```

Comment:
One final thing.
You have two sets of lists here, each representing the icon states in a dmi.
Is there any reason you couldn't just have one set of lists, where each list represents two dmis?

Reply:
Mainly because I've already told golden about 5 times that they are finally free to review it and I don't want to make any more changes that might need a new review.

But yeah theres no real reason. I might fix this later, in another pr.

Reply:
(or maybe just in this pr who the hell knows)

Reply:
I'm almost done reviewing everything, go ahead and make that change, and I think we'd be good for at least a test-merge.